external-crates/move/move-compiler/src/shared/mod.rs

```diff
@@ -235,6 +264,25 @@ impl CompilationEnv {
         final_diags
     }
 
+    /// Add a new filter for warnings
+    pub fn add_warning_filter_scope(&mut self, mut filter: WarningFilters) {
+        // This essentially "clones" the current filter into the next scope. This should be
+        // efficient enough since the diag_filter vec should be only about 2 or 3 elements deep
+        // and the size of the filter should only be relatively small (at most 10 or so elements)
+        debug_assert!(
+            self.warning_filter.len() <= 3,
+            "TODO If triggered this TODO you might want to make this more efficient"
+        );
```

Comment:
Mmm not sure how I feel about this assert. I would prefer if it were removed as I don't think it is really adding anything useful other than a place for developers to stub their toes.